### PR TITLE
Fix neutral town unit growth

### LINKED ISSUE #5905: Neutral town creature growth shouldn't be affected by difficulty
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

On a start of 4 week playing on normal:
image

On a start of 4 week playing on impossible:
image

In the OG creature growth doesn't affected by game difficulty. So it's a half of the weekly castle growth.

### Save file

[save.zip](https://github.com/ihhub/fheroes2/files/9684924/save.zip)

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by ihhub**
The growth should not be affected by game's difficulty. close #5905

### CODE REVIEW

**Comment by ihhub** [comment]
@Branikolog , please test these changes.

**Review by zenseii — COMMENTED** [review]
Hi @ihhub,

I did some tests and the creature dwelling growth for neutral cities is now correct. The only exception is the case noted by @Branikolog in week 5 when the growth is doubled.

Here's my numbers:
<details>

<summary>Growth numbers</summary>
easy

week 1
peasant = 12
archer = 8

week 2
peasant = 18
archer = 12

week3 
peasant = 24
archer = 16

week 4
peasant = 30
archer = 20

week 5
peasant = 42
archer = 28

normal

week 1
peasant = 12
archer = 8

week 2
peasant = 18
archer = 12


</details>

**Review by Branikolog — APPROVED** [review]
Hi, @ihhub .
Growth is no more affected by difficulty.
But @zenseii is right, 5th week is still increased.

File: src/fheroes2/castle/castle.cpp [hunk 1]
```diff
@@ -564,8 +564,9 @@ void Castle::ActionNewWeek()
                 if ( ( dwellings1[ii] == DWELLING_MONSTER1 ) && ( building & BUILD_WEL2 ) )
                     growth += GetGrownWel2();
 
-                if ( isControlAI() )
+                if ( isControlAI() && !isNeutral ) {
                     growth = static_cast<uint32_t>( growth * Difficulty::GetUnitGrowthBonusForAI( Game::getDifficulty() ) );
+                }
 
                 // neutral town: half population (normal for begin month)
                 if ( isNeutral && !world.BeginMonth() )
```
Comment (oleg-derevenetz) [inline src/fheroes2/castle/castle.cpp:572]: It seems that for neutral towns the growth modifier is always 0.5, even in the beginning of a new month, so I believe that we should remove `&& !world.BeginMonth()` from here (and update the comment accordingly).

**Review by oleg-derevenetz — CHANGES_REQUESTED** [review]
Hi @ihhub I just left a proposal regarding the 5th week, please take a look when you have time.

File: src/fheroes2/castle/castle.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/castle/castle.cpp]: Fixed.

**Comment by ihhub** [comment]
Hi @zenseii , @Branikolog and @oleg-derevenetz , I fixed growth for neutral towns at the beginning of a month. Please take a look.

**Review by oleg-derevenetz — APPROVED** [review]

**Review by zenseii — APPROVED** [review]

**Pull request merged**